extensions/amazon-lambda-resteasy/maven/src/main/java/io/quarkus/amazon/lambda/resteasy/maven/GenerateSamConfigMojo.java

```diff
@@ -0,0 +1,120 @@
+package io.quarkus.amazon.lambda.resteasy.maven;
+
+import java.io.File;
+import java.util.List;
+
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.Component;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.apache.maven.plugins.annotations.ResolutionScope;
+import org.apache.maven.project.MavenProject;
+import org.apache.maven.project.MavenProjectHelper;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.repository.RemoteRepository;
+
+import io.quarkus.bootstrap.model.AppArtifact;
+import io.quarkus.bootstrap.model.AppModel;
+import io.quarkus.bootstrap.resolver.BootstrapAppModelResolver;
+import io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver;
+import io.quarkus.bootstrap.resolver.maven.workspace.LocalProject;
+import io.quarkus.creator.AppCreator;
+import io.quarkus.creator.AppCreatorException;
+import io.quarkus.creator.phase.curate.CurateOutcome;
+import io.quarkus.maven.utilities.MojoUtils;
+
+@Mojo(name = "configure-sam", defaultPhase = LifecyclePhase.GENERATE_RESOURCES, requiresDependencyResolution = ResolutionScope.COMPILE_PLUS_RUNTIME)
+public class GenerateSamConfigMojo extends AbstractMojo {
```

Comment:
This is probably ok for now, but we need to move this sort of thing into the augment phase as much as possible at some point in the future.

Reply:
TBH I think every thing we make a new Mojo for is going to make it that much harder to move to a build phase.  And maybe I'm wrong but I am not sure I see a reason why this can't be done as build steps?

Reply:
I would love it to be a build step.  generating output to arbitrary locations gets tricky in ways that aren't maven specific

Reply:
moved to a build step as @dmlloyd suggested.  not entirely sure it's "better" but maybe it is.  it's at least build agnostic.